pkg/collector/corechecks/servicediscovery/service_detector_test.go

```diff
@@ -0,0 +1,32 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package servicediscovery
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/language"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/usm"
+)
+
+func TestFixup(t *testing.T) {
+	meta := fixupMetadata(usm.ServiceMetadata{Name: "Foo", DDService: "Bar"}, language.Go)
```

Comment:
nit - to cover more cases (and not handling only first letter)
```suggestion
	meta := fixupMetadata(usm.ServiceMetadata{Name: "fOo", DDService: "BAR"}, language.Go)
```